### PR TITLE
Allow introspection into ReentrantMutex

### PR DESCRIPTION
**Pull request by drcapulet**
Required for rspec/rspec-core#2340.

### CODE REVIEW

File: lib/rspec/support/reentrant_mutex.rb [hunk 1]
```diff
@@ -25,10 +25,18 @@ def synchronize
         exit
       end
 
+      def locked?
+        @owner != nil
+      end
+
+      def owned?
+        @owner == Thread.current
+      end
```
Comment (drcapulet) [inline lib/rspec/support/reentrant_mutex.rb]: Updated, let me know if you want anything more complex.

Comment (JonRowe) [inline lib/rspec/support/reentrant_mutex.rb]: Great, as per @joshcheek's comments on rspec/rspec-core#2340 we still need these methods on our own version of mutex for 1.8.7 too :)

Comment (drcapulet) [inline lib/rspec/support/reentrant_mutex.rb]: `ReentrantMutex` is a wrapper on top of either the native Ruby `Mutex` class of the RSpec version ([initialization](https://github.com/rspec/rspec-support/blob/master/lib/rspec/support/reentrant_mutex.rb#L18) and [definition](https://github.com/rspec/rspec-support/blob/master/lib/rspec/support/reentrant_mutex.rb#L44-L51)) so this should be the only place changes are required.

**Comment by JonRowe** [comment]
Thanks :)

**Pull request merged**